examples/classification_example/main_train.py

```diff
@@ -0,0 +1,266 @@
+    embedding_dim = list(embed_dict.values())[0].shape[-1]
+
+    scale = np.sqrt(3.0 / embedding_dim)
+    table = np.empty(
+        [extractor.size(), embedding_dim], dtype=np.float32
+    )
+    oov = 0
+    for word, index in extractor.items():
+        if word in embed_dict:
+            embedding = embed_dict[word]
+        elif word.lower() in embed_dict:
+            embedding = embed_dict[word.lower()]
+        else:
+            embedding = np.random.uniform(
+                -scale, scale, [1, embedding_dim]
+            ).astype(np.float32)
+            oov += 1
+        table[index, :] = embedding
+    return torch.from_numpy(table)
+
+
+def create_model(text_extractor: AttributeExtractor,
+                 config: Config, in_channels: int):
+    embedding_dict = {}
+    fake_tensor = torch.tensor([0.0 for i in range(100)])
+    for word, index in text_extractor.items():
+        embedding_dict[word] = fake_tensor
+
+    word_embedding_table = \
+        construct_word_embedding_table(embedding_dict, text_extractor)
+
+    model: nn.Module = \
+        CNN_Classifier(in_channels=in_channels,
+                       word_embedding_table=word_embedding_table)
+
+    if config.config_model.model == "bert":
+        model: nn.Module = BERTClassifier()
+
+    return model, word_embedding_table
+
+
+def train(model: nn.Module, optim: Optimizer, batch: Batch, max_sen_length: int):
+    word = batch["text_tag"]["data"]
+    labels = batch["label_tag"]["data"]
+    optim.zero_grad()
+
+    logits, pred = None, None
+
+    if config.config_model.model == "cnn":
+        logits, pred = model(batch)
+
+    if config.config_model.model == "bert":
+        logits, pred = model(pad_each_bach(word, max_sen_length))
+
+    labels_1D = torch.squeeze(labels)
+
+    true_one_batch = (labels_1D == pred).sum().item()
+
+    loss = criterion(logits, labels_1D)
+
+    loss.backward()
+    optim.step()
+
+    batch_train_err = loss.item() * batch.batch_size
+
+    return batch_train_err, true_one_batch
+
+
+# All the configs
+config_data = yaml.safe_load(open("config_data.yml", "r"))
+config_model = yaml.safe_load(open("config_model.yml", "r"))
+
+config = Config({}, default_hparams=None)
+config.add_hparam('config_data', config_data)
+config.add_hparam('config_model', config_model)
+
+# Generate request
+text_extractor: AttributeExtractor = \
+    AttributeExtractor(config={"entry_type": Token,
+                               "vocab_method": "indexing",
+                               "attribute": "text"})
+
+
+class SentimentExtractor(AttributeExtractor):
+    def get_attribute(self, entry: Entry, attr: str):
+        if "positive" in getattr(entry, attr):
+            return "positive"
+        else:
+            return "negative"
+
+    def set_attribute(self, entry: Entry, attr: str, value: Any):
+        if value == "positive":
+            getattr(entry, attr)["positive"] = 1.0
+        else:
+            getattr(entry, attr)["negative"] = 0.0
```

Comment:
```suggestion
    entry.sentiment = {
       "positive": 1.0,
       "negative": 0.0,
    }
```
Here you are implementing a very specific extractor for this case only, you don't need to use `getattr`

Reply:
This is Yanwen's advice, to overwrite get_attribute and set_attribute. I think this is good because this gives an example for how to write an extended extractor that is user defined. In this case, you can make it short without using get_attribute, but if the condition is more complicated, the user is better to know he can also define get_attribute.

Reply:
This is not a good idea. If the condition is complicated, the user will know what to do. Not using `getattr` is not because we want shorter code, but we want to provide intuitive examples.

Reply:
I never have such advice. I think @ziqian98 misunderstood what @hunterhector said. @ziqian98 is talking about overwrite extractor methods: `get_attribute` and `set_attribute`. And @hunterhector is talking about how to reference the `sentiment` dict in the `Sentence` class. Basically, we should not use `getattr` to reference the `sentiment` as this is user-side code. @ziqian98 You should follow what Hector said here in the suggested change.

Reply:
Sorry, I misunderstand your idea. I confuse the get_attribute with getattr. I will fix it right away

Reply:
Modified in the new commit.